### PR TITLE
Profile merge regressions

### PR DESCRIPTION
**Pull request by drewdahlke**
I ramped up our unit test coverage on profile merging and found several bugs in the process

### CODE REVIEW

File: core/src/test/java/com/whylogs/core/DatasetProfileTest.java [hunk 1]
```diff
@@ -261,6 +256,54 @@ public void deserialization_withRegressionData() throws IOException {
...
+  public void testMergeRecentWithOlderProfileOppositeDirection() throws IOException {
+    val profile1 = DatasetProfile.parse(getClass().getResourceAsStream("/python_profile.bin"));
+    val profile2 = DatasetProfile.parse(getClass().getResourceAsStream("/regression.bin"));
+    val merged = profile2.merge(profile1);
+    assertNotNull(merged.getModelProfile());
+    assertNotNull(merged.getModelProfile().getMetrics());
+    asertMetrics(merged.getModelProfile().getMetrics().getRegressionMetrics(), 1);
+  }
+
+  private void asertMetrics(RegressionMetrics metrics, int multiplier) {
```
Comment (cswarth) [inline core/src/test/java/com/whylogs/core/DatasetProfileTest.java]: typo "assertMetrics"

**Comment by cswarth** [comment]
LGTM 

Might consider whether unit test column_Merge_StringTracker adds any value now.

**Review by cswarth — DISMISSED** [review]

File: core/src/main/java/com/whylogs/core/DatasetProfile.java [hunk 2]
```diff
@@ -290,12 +290,14 @@ private DatasetProfile doMerge(@NonNull DatasetProfile other, Map<String, String
       result.columns.put(column, thisColumn.merge(otherColumn));
     }
 
+    result.modelProfile = this.modelProfile;
```
Comment (andyndang) [inline core/src/main/java/com/whylogs/core/DatasetProfile.java]:
```suggestion
    result.modelProfile = this.modelProfile.copy();
```

Comment (andyndang) [reply core/src/main/java/com/whylogs/core/DatasetProfile.java]: FYI. Need to copy here since `merge` returns a new object

**Review by andyndang — APPROVED** [review]
LGTM

**Pull request merged**